src/hotspot/share/gc/shenandoah/shenandoahGenerationalHeap.cpp

```diff
@@ -847,7 +848,7 @@ class ShenandoahGenerationalUpdateHeapRefsTask : public WorkerTask {
       r = _regions->next();
     }
 
-    if (!gc_generation->is_global()) {
+    if (!_generation->is_global()) {
```

Comment:
Is this ever `_generation->is_old()` ? I am guessing this is either global or young? I am guessing for the case of update_refs for a mixed collection we still pass in the `young` gen here?

It might be a good idea wherever we have a `_generation` field member in a class to indicate what it represents. In most cases, it's probably clear that it represents the generation subject to collection in that cycle. But for mixed collection sets, what does the `_generation` field represent?